python/rtypes/pcc/triggers.py

```diff
@@ -0,0 +1,35 @@
+
+
+class Trigger(object):
+
+    def __init__(self, pcc_type, time, action, priority=None):
+        self.pcc_type = pcc_type
+        self.time = time
+        self.action = action
+        self.priority = priority
+
+    def __call__(self, procedure):
+
+        class TriggerProcedure(object):
+
+            def __init__(self, procedure):
+                self.procedure = procedure
+
+            def __call__(self):
+                return self.procedure
```

Comment:
This would be wrong. You would need to pass in args and kwargs to call, and actually call the self.procedure function within __call__, not just return the function itself.

Reply:
if too tedious for now and you will refactor later, put them all under 
```
if __name__ == "__main__":
  testing blah blah blah.
```